src/save.js

```diff
@@ -0,0 +1,89 @@
+const Promise = require('bluebird'),
```

Comment:
Advanced Promise management enabling creation of Promises based on callback functions and asynchronous return values.